packages/peregrine/lib/talons/CartPage/PriceAdjustments/ShippingMethods/useShippingMethods.js

```diff
@@ -74,3 +88,32 @@ export const useShippingMethods = props => {
         showForm
     };
 };
+
+/** JSDocs type definitions */
+
+/**
+ * GraphQL queries for shipping methods.
+ * This is a type used in the {@link useShippingMethods} talon.
+ *
+ * @typedef {Object} ShippingMethodsQueries
+ *
+ * @property {GraphQLAST} getShippingMethodsQuery Query to get the available shipping methods.
+ *
+ * @see [shippingMethods.gql.js]{@link https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/CartPage/PriceAdjustments/ShippingMethods/shippingMethods.gql.js}
+ * for the queries used in Venia
+ */
+
+/**
+ * Object type returned by the {@link useShippingMethods} talon.
+ * It provides prop data to use when rendering shipping methods.
+ *
+ * @typedef {Object} ShippingMethodsProps
+ *
+ * @property {number} hasMethods Provides the number of shipping methods available.
+ * Can be used as a boolean value since having no shipping methods would return 0.
+ * @property {boolean} isShowingForm True if the form should be shown. False otherwise.
+ * @property {SelectShippingFields} selectedShippingFields Values for the select input fields on the shipping form
```

Comment:
Yes and no. This is my attempt at being clever. Since I knew this was going to be grouped with the other ShippingMethods talons, I just referenced the `SelectShippingFields` defined in the `useShippingForm()` talon.
If I listed it here, it would list it twice on the generated page.